Set version range in manifest, up to local swift version to resolve

SixFiveSoftware opened the pull request:
Here's the PR to set the version to a range. It looks like depending on your local Swift version, it will pick the version compatible with your current toolchain, according to their [GitHub repo](https://github.com/swiftlang/swift-syntax?tab=readme-ov-file#releases).

SixFiveSoftware committed:
Set version range in manifest, up to local swift version to resolve

bjsaturn committed:
Bump to Swift 6.0 and up major

SixFiveSoftware committed:
Merge pull request #1 from bjsaturn/bj/version-range-p2

Bump to Swift 6.0 and up major

JosephDuffy committed:
Merge branch 'main' into bj/version-range

JosephDuffy committed:
Use same version range in Package@swift-5.9.swift

JosephDuffy committed:
Test swift-sytnax range on CI

JosephDuffy committed:
Merge branch 'main' into bj/version-range

JosephDuffy reviewed (approved):
Just need to check CI works so may push more commits.

JosephDuffy committed:
Run tests on pull request target

JosephDuffy committed:
Pin swift-dependency-compatibility-checker to main branch

JosephDuffy committed:
Allow Codebcov upload to fail for PRs from forks

JosephDuffy committed:
Remove swift-dependency-compatibility-checker from Mintfile

JosephDuffy committed:
Skip Codecov upload for PRs from forks

JosephDuffy committed:
Set Package.resolved version to 2

JosephDuffy committed:
Remove support for Swift < 6.0

JosephDuffy committed:
Silent output for mint to enable parsing as matrix

JosephDuffy committed:
Check event name for pull request target, not `event.action`

JosephDuffy committed:
Revert Package.resolved changes

JosephDuffy committed:
Remove testing of Swift < 6.0

JosephDuffy committed:
Use codecov-actions@v5, which should support uploading from forks

Maybe it will need `use_oidc`?

JosephDuffy committed:
Use `files`, which is the _replacement_, not the deprecated argument

JosephDuffy committed:
Select Xcode 26.0.1, install mint

JosephDuffy committed:
Include macOS version in cache keys

JosephDuffy committed:
Update xcbeautify to 3.1.2

1.0.1 is very old. Support for parsing `swift test` output (or specifically Swift Testing?) was in 2.12.0.

JosephDuffy committed:
Use xcbeautify 2.30.1 to support Swift 6.0

It actually support Swift 5.9, but we just need down to 6.0 right now

JosephDuffy committed:
Fail job when tests fail

JosephDuffy committed:
Fix tests with Swift Syntax 510.0.0 or greater

JosephDuffy committed:
Fix warning on Swift Syntax 600.0.0

JosephDuffy committed:
Fix expected output on Swift Syntax 600.0.0

JosephDuffy committed:
Fix test assertions on Swift Syntax 602.0.0

JosephDuffy committed:
Only compile against Swift Syntax 601.0.0

JosephDuffy committed:
Use single quotes to fix syntax error

JosephDuffy committed:
Replace deprecated inputs

JosephDuffy committed:
Always depend on HashableMacroFoundation

JosephDuffy committed:
Fix step name

JosephDuffy committed:
Upload coverage for all Swift Syntax versions

JosephDuffy committed:
Enable code coverage
